test/smoketest/requirements.in

```diff
@@ -0,0 +1,8 @@
+# "Abstract" dependencies for the project
+#
+# The sibling requirements.txt file contains the corresponding concrete /
+# pinned dependencies.  You can use pip-tools to automatically update
+# requirements.txt from this file:
+# https://github.com/jazzband/pip-tools
```

Comment:
`pip-tools` wouldn't be mandatory. It would only be useful to / potentially used by someone who wanted to update requirements.txt, e.g. to check for upgrades in package dependencies and/or when adding a new abstract dependency.

One possible "manual" approach to updating requirements.txt would be--

1. Create a fresh virtualenv
2. Manually invoke `pip install ...` with whatever abstract dependencies you want (e.g. requests, etc).
3. Write the output of `pip freeze` to a file (e.g. `pip freeze > requirements.txt`.

(This is essentially what pip-tools does.)

One reason I mentioned `pip-tools` is that it lets you specify the abstract dependencies in a stand-alone `requirements.in` file, which is nice. I'm not sure other tools like pip recognize such a file format (though they might).